crypto/ecies/ecies_test.go

```diff
@@ -143,6 +148,55 @@ func TestSharedKey(t *testing.T) {
 	}
 }
 
+func TestSharedKeyPadding(t *testing.T) {
+	// TODO: remove after refactoring packages crypto and crypto/ecies
+	GetKey := func(prv []byte) *PrivateKey {
+		priv := new(ecdsa.PrivateKey)
+		priv.PublicKey.Curve = secp256k1.S256()
+		priv.D = common.BigD(prv)
+		priv.PublicKey.X, priv.PublicKey.Y = secp256k1.S256().ScalarBaseMult(prv)
+		return ImportECDSA(priv)
+	}
+
+	// sanity checks
+	prv0 := GetKey(common.FromHex("1adf5c18167d96a1f9a0b1ef63be8aa27eaf6032c233b2b38f7850cf5b859fd9"))
+	prv1 := GetKey(common.FromHex("97a076fc7fcd9208240668e31c9abee952cbb6e375d1b8febc7499d6e16f1a"))
```

Comment:
Would prefer to avoid the dependency on package common just for hex parsing.

```go
GetKey := func(prv string) *PrivateKey {
    ...
    priv.D, _ = new(big.Int).SetString(prv, 16)
    ...
}
```
}